controller/iteration_blackbox_test.go

```diff
@@ -1080,3 +1080,193 @@ func (rest *TestIterationREST) TestIterationDelete() {
 		}
 	})
 }
+
+func (rest *TestIterationREST) TestUpdateIteration() {
+	resetFn := rest.DisableGormCallbacks()
+	defer resetFn()
+	rest.T().Run("update success - iteration parent", func(t *testing.T) {
+		// build following structure
+		// 	root 0
+		// 		|---- itr 1
+		//			|---- itr 2
+		//				|---- itr 3
+		//  				|---- itr 4
+		// 			|---- itr 5
+		// 		|---- itr 6
+
+		// given
+		fxt := tf.NewTestFixture(t, rest.DB,
+			tf.Iterations(7, tf.SetIterationNames("root iteration", "iteration 1",
+				"iteration 2", "iteration 3", "iteration 4", "iteration 5", "iteration 6"),
+				func(fxt *tf.TestFixture, idx int) error {
+					itr := fxt.Iterations[idx]
+					switch idx {
+					case 1:
+						itr.MakeChildOf(*fxt.Iterations[0])
+					case 2:
+						itr.MakeChildOf(*fxt.Iterations[1])
+					case 3:
+						itr.MakeChildOf(*fxt.Iterations[2])
+					case 4:
+						itr.MakeChildOf(*fxt.Iterations[3])
+					case 5:
+						itr.MakeChildOf(*fxt.Iterations[1])
+					case 6:
+						itr.MakeChildOf(*fxt.Iterations[0])
+					}
+					return nil
+				}))
+		svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
+		itr1 := fxt.IterationByName("iteration 1")
+		itr2 := fxt.IterationByName("iteration 2")
+		itr3 := fxt.IterationByName("iteration 3")
+		itr4 := fxt.IterationByName("iteration 4")
+		itr5 := fxt.IterationByName("iteration 5")
+
+		// update parent of iteration 3 (move itr3 under itr1)
```

Comment:
Only moving iteration 3 is boring because it is a leaf iteration without any children.

Reply:
actually iteration 3 is not leaf node, it is parent of iteration 4 and child to the iteration 2.
I have used "IterationByName" now and should be more readable https://github.com/fabric8-services/fabric8-wit/pull/1770/commits/accdb037340d51a5d0a9c5f852b382a8498846ef

Reply:
Okay I see now, but then you have to update the documentation above because it shows iteration 4 as a sibling to iteration 3:

```go
 		// build following structure
 		// 	root 0
 		// 		|---- itr 1
 		//			|---- itr 2
 		//				|---- itr 3
 		//  				|---- itr 4
 		// 			|---- itr 5
 		// 		|---- itr 6
```